indy_node/test/anon_creds/helper.py

```diff
@@ -1,25 +1,36 @@
+from plenum.common.types import f
+
 from plenum.common.constants import STATE_PROOF, ROOT_HASH, MULTI_SIGNATURE, PROOF_NODES, MULTI_SIGNATURE_SIGNATURE, \
     MULTI_SIGNATURE_PARTICIPANTS, MULTI_SIGNATURE_VALUE, MULTI_SIGNATURE_VALUE_LEDGER_ID, \
     MULTI_SIGNATURE_VALUE_STATE_ROOT, MULTI_SIGNATURE_VALUE_TXN_ROOT, MULTI_SIGNATURE_VALUE_POOL_STATE_ROOT, \
-    MULTI_SIGNATURE_VALUE_TIMESTAMP, TYPE, DATA
+    MULTI_SIGNATURE_VALUE_TIMESTAMP, TYPE, DATA, TXN_TIME, TXN_PAYLOAD, TXN_PAYLOAD_METADATA, TXN_PAYLOAD_METADATA_FROM, \
+    TXN_PAYLOAD_DATA, TXN_METADATA, TXN_METADATA_SEQ_NO, TXN_METADATA_TIME
 from indy_common.constants import VALUE, GET_REVOC_REG_DEF, GET_REVOC_REG, GET_REVOC_REG_DELTA, \
-    ACCUM_FROM, ACCUM_TO, STATE_PROOF_FROM, ISSUED
+    ACCUM_TO, ISSUED, STATE_PROOF_FROM, ACCUM_FROM
 from common.serializers.serialization import state_roots_serializer, proof_nodes_serializer
 from state.pruning_state import PruningState
 from indy_common.state import domain
 
 
+def add_txn_specific_fileds(result):
+    result[TXN_PAYLOAD] = {TXN_PAYLOAD_METADATA: {TXN_PAYLOAD_METADATA_FROM: result[f.IDENTIFIER.nm]}}
+    result[TXN_PAYLOAD][TXN_PAYLOAD_DATA] = result[TXN_PAYLOAD_DATA]
+    result[TXN_METADATA] = {TXN_METADATA_SEQ_NO: result[TXN_METADATA_SEQ_NO]}
+    result[TXN_METADATA][TXN_METADATA_TIME] = result[TXN_METADATA_TIME]
+
+
 def prepare_for_state(result):
     request_type = result[TYPE]
+    add_txn_specific_fileds(result)
     if request_type == GET_REVOC_REG_DEF:
-        return domain.prepare_revoc_def_for_state(result['data'])
+        return domain.prepare_revoc_def_for_state(result)
```

Comment:
Why don't we call `prepare_get_revoc_def_for_state`? `result` is a result of get request, but `prepare_revoc_def_for_state` assumes `txn` as an argument.